src/DuplexResourceStream.php

```diff
@@ -113,7 +106,9 @@ public function pause()
     public function resume()
     {
         if (!$this->listening && $this->readable) {
-            $this->loop->addReadStream($this->stream, array($this, 'handleData'));
+            $this->loop->addReadStream($this->stream, function () {
+                $this->handleData();
+            });
```

Comment:
While I agree this looks *cleaner*, I would suggest keeping the original array in this PR to avoid using an additional closure (stack frame / overhead):

```suggestion
            $this->loop->addReadStream($this->stream, [$this, 'handleData']);
```

That said, I'm happy to discuss this in a separate PR to evaluate this and its potential impact in more depth.

Reply:
> That said, I'm happy to discuss this in a separate PR to evaluate this and its potential impact in more depth.

Sounds like a plan :+1: